src/Monomer/Widgets/Animation/Transform.hs

```diff
@@ -0,0 +1,284 @@
+{-|
+Module      : Monomer.Widgets.Animation.Transform
```

Comment:
@Deltaspace0 could this one have a small example, similar to [this](https://github.com/fjvallarino/monomer/blob/main/src/Monomer/Widgets/Singles/TextField.hs#L12)?

Reply:
```haskell
animTransform f managed where
  f t (Rect x y w h) =
    [ animTranslation $ Point (ft t w) (ft t h)
    , animScale $ Point (fs t) (fs t)
    ]
```

This is from `Zoom` widget with no configurations. Is it good?

Reply:
```haskell

@
animTransform f managed where
  f t (Rect x y w h) =
    [ animTranslation $ Point (ft t w) (ft t h)
    , animScale $ Point (fs t) (fs t)
    ]
@

With configuration options:

@
animTransform_ [duration 2000, autoStart] f managed where
  f t (Rect x y w h) =
    [ animTranslation $ Point (ft t w) (ft t h)
    , animScale $ Point (fs t) (fs t)
    ]
@
```
It is small enough and shows that multiple transformations can be combined.

Reply:
Maybe something like this?

```haskell
transform f t (Rect x y w h) = [
  animTranslation $ Point (ft t w) (ft t h),
  animScale $ Point (fs t) (fs t)
]

animTransform transform managed
```

but instead of using `f` and `t` use a more descriptive name (I assume `t` is time and it's `ts` in other places, but I'm not sure what `f` represents). I guess `ft` and `fs` should also be part of the example.

Reply:
@Deltaspace0 I received your second example after I posted. They are both good, the only thing I would suggest is the second part of my comment:

Instead of using `f` and `t` use a more descriptive name (I assume `t` is time and it's `ts` in other places, but I'm not sure what `f` represents). I guess `ft` and `fs` should also be part of the example.

Reply:
`f` represents `f`unction and `ft` and `fs` are functions which return `t`ranslation and `s`cale coordinates respectively.

```haskell

@
transform t (Rect x y w h) =
  [ animTranslation $ Point tx ty
  , animScale $ Point sx sy
  ]

animTransform transform managed
@

With configuration options:

@
transform t (Rect x y w h) =
  [ animTranslation $ Point tx ty
  , animScale $ Point sx sy
  ]

animTransform_ [duration 2000, autoStart] transform managed
@
```

Here I simplified it, should be more readable (though in real example `tx`, `ty` etc should depend on `t` and `Rect`)

Reply:
@Deltaspace0 I think keeping those functions that depend on time is better since users will end up writing code similar to that. Including the definitions for `ft` and `fs` in the example will be clear enough.

Sorry if this sounds like nitpicking.

Reply:
I am just afraid that this example will turn out to be full implementation of the `Zoom` widget, and it is supposed to only show how to use the widget.

```haskell
@
fs t = clamp 0 1 $ t/500
ft t s = (1-(fs t))*s/2

transform t (Rect x y w h) =
  [ animTranslation $ Point (ft t w) (ft t h)
  , animScale $ Point (fs t) (fs t)
  ]

animTransform transform managed
@
```
Here I included `fs` and `ft`, but I think that it is more confusing than just `tx`, `ty` etc

Reply:
@Deltaspace0 fair enough. The version with just `tx` and `ty` is good; maybe mention that a more complex example can be found in Zoom (similar to how it's done [here](https://github.com/fjvallarino/monomer/blob/main/src/Monomer/Widgets/Containers/Base/LabeledItem.hs#L12)).

Reply:
I added links to `Zoom` and `Shake` widgets, since they both use multiple transformations.